src/librustdoc/core.rs

```diff
@@ -82,7 +82,8 @@ pub struct DocContext<'a, 'tcx: 'a, 'rcx: 'a> {
     pub fake_def_ids: RefCell<FxHashMap<CrateNum, DefId>>,
     pub all_fake_def_ids: RefCell<FxHashSet<DefId>>,
     /// Maps (type_id, trait_id) -> auto trait impl
-    pub generated_synthetics: RefCell<FxHashSet<(DefId, DefId)>>
+    pub generated_synthetics: RefCell<FxHashSet<(DefId, DefId)>>,
+    pub current_item_id: RefCell<Option<Name>>,
```

Comment:
We tried so hard to keep from having to use a `current_item` field in the `DocContext`, but it looks like this is the easiest way, short of not processing attributes through the `Clean` trait.

Could you make this say `current_item_name`, since it's not really an ID we're putting here?

Reply:
Oh true, I changed meanwhile and forgot to update the variable's name.